ZipBuilder/Sources/ZipBuilder/FrameworkBuilder.swift

```diff
@@ -345,7 +371,246 @@ struct FrameworkBuilder {
+  /// and thin archives for each architecture slice.
+  /// - Parameter fromFolder: The almost complete framework folder. Includes everything but the binary.
+  /// - Parameter thinArchives: All the thin archives.
+  private func packageXCFramework(withName framework: String,
+                                  fromFolder: URL,
+                                  thinArchives: [Architecture: URL],
+                                  moduleMapContents: String) -> URL {
+    let fileManager = FileManager.default
+
+    // Create a `.framework` for each of the thinArchives using the `fromFolder` as the base.
+    let platformFrameworksDir =
+      fileManager.temporaryDirectory(withName: "platform_frameworks")
+    if !fileManager.directoryExists(at: platformFrameworksDir) {
+      do {
+        try fileManager.createDirectory(at: platformFrameworksDir,
+                                        withIntermediateDirectories: true)
+      } catch {
+        fatalError("Could not create a temp directory to store all thin frameworks: \(error)")
+      }
+    }
+
+    // Group the thin frameworks into three groups: device, simulator, and Catalyst (all represented
+    // by the `TargetPlatform` enum. The slices need to be packaged that way with lipo before
+    // creating a .framework that works for similar grouped architectures. If built separately,
+    // `-create-xcframework` will return an error and fail:
+    // `Both ios-arm64 and ios-armv7 represent two equivalent library definitions`
+    var frameworksBuilt: [URL] = []
+    for platform in Architecture.TargetPlatform.allCases {
+      // Get all the slices that belong to the specific platform in order to lipo them together.
+      let slices = thinArchives.filter { $0.key.platform == platform }
+      let platformDir = platformFrameworksDir.appendingPathComponent(platform.rawValue)
+      do {
+        try fileManager.createDirectory(at: platformDir, withIntermediateDirectories: true)
+      } catch {
+        fatalError("Could not create directory for architecture slices on \(platform) for " +
+          "\(framework): \(error)")
+      }
+
+      // Package a normal .framework with the given slices.
+      let destination = platformDir.appendingPathComponent(fromFolder.lastPathComponent)
+      packageFramework(withName: framework,
+                       fromFolder: fromFolder,
+                       thinArchives: slices,
+                       destination: destination,
+                       moduleMapContents: moduleMapContents)
+
+      frameworksBuilt.append(destination)
+    }
+
+    // We now need to package those built frameworks into an XCFramework.
+    let xcframeworksDir = projectDir.appendingPathComponent("xcframeworks")
+    if !fileManager.directoryExists(at: xcframeworksDir) {
+      do {
+        try fileManager.createDirectory(at: xcframeworksDir,
+                                        withIntermediateDirectories: true)
```

Comment:
Nit: this can likely fit on the line above.

Reply:
Nope. 101 characters.